osu.Game/IO/StableStorage.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using osu.Framework.Platform;
+
+namespace osu.Game.IO
+{
+    /// <summary>
+    /// A storage pointing to an osu-stable installation.
+    /// Provides methods for handling installations with a custom Song folder location.
+    /// </summary>
+    public class StableStorage : DesktopStorage
+    {
+        private const string stable_default_songs_path = "Songs";
+
+        private readonly DesktopGameHost host;
+        private readonly Lazy<string> songsPath;
+
+        public StableStorage(string path, DesktopGameHost host)
+            : base(path, host)
+        {
+            this.host = host;
+            songsPath = new Lazy<string>(locateSongsDirectory);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Storage"/> pointing to the osu-stable Songs directory.
+        /// </summary>
+        public Storage GetSongStorage() => new DesktopStorage(songsPath.Value, host);
+
+        private string locateSongsDirectory()
+        {
+            var songsDirectoryPath = Path.Combine(BasePath, stable_default_songs_path);
+
+            var configFile = GetFiles(".", "osu!.*.cfg").FirstOrDefault();
+
+            if (configFile == null)
+                return songsDirectoryPath;
+
+            using (var textReader = new StreamReader(GetStream(configFile)))
```

Comment:
other usages put both the stream get and the stream reader in `using` blocks:

https://github.com/ppy/osu/blob/67014018fc7b3f6a7bd3ba53ff6660ec9b972a7f/osu.Game.Tournament/IPC/FileBasedIPC.cs#L77-L79

it is perhaps a bit superfluous as looking at the implementation of `StreamReader` i think it'll clean up the underlying stream, but i'm not sure that's safe to rely on.

Reply:
`StreamReader` does dispose of the underlying stream by default unless it threw an exception during construction, so yeah I should probably wrap up the stream in a using statement aswell just in case.